libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/CriDeserializer.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.di.ipv.core.library.domain;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.io.IOException;
+
+public class CriDeserializer extends StdDeserializer<Cri> {
+
+    public CriDeserializer() {
+        this(null);
+    }
+
+    public CriDeserializer(Class<Cri> c) {
+        super(c);
+    }
+
+    @Override
+    public Cri deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        return Cri.fromId(p.getText());
```

Comment:
Does this need to handle `null`? I notice that the `CriCheckingService` checks for a null/empty value. It may well be better to throw here (if it's an error anyway), but worth making sure we get a useful error message.

(We could also handle `null` in `Cri.fromId`, but that might itself be confusing!)

Reply:
We should get an IllegalArgumentException if its null, but can change to something that must be caught and handled instead?

Reply:
I'm not too fussed, so long as the message makes it obvious what's gone wrong (i.e. is not just `NullPointerException: null`).

If you're moving it to a separate method, you could just make that method return `null` if the `credentialIssuerId` is `null` and rely on the existing validationg logic.